sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -687,6 +687,7 @@ def _fit_stochastic(
                         % self.max_iter,
                         ConvergenceWarning,
                     )
+                    break
```

Comment:
Actually, looking at the code, I think it would make more sense to change the for loop from:

```python
for it in range(self.max_iter):
```

to:

```python
for it in range(self.max_iter - self.n_iter_):
```

Basically, it is incorrect for the `for` loop to iterate `max_iter` at each call. We only want to have the difference when warm-starting. It looks more logical to me than breaking the loop.

Reply:
Hey, thanks for review. I did some changes in the `test_mlp.py` and don't understand why the test `test_learning_rate_warmstart()` divide `mlp.learning_rate_init` through `pow()`. I did the change and it was pass but I can not explain why it works. Have you an answer for me?

The other changes made sense for me, but please have a look.